mutations: do not generate histograms for JSON

otan opened the pull request:
EncodeTableKey does not work for JSON, so avoid collecting histograms
for them.

Resolves #62057
Resolves #62058

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62127)
<!-- Reviewable:end -->

yuzefovich commented:
I added #62057 to be resolved by this PR too.

otan committed:
mutations: do not generate histograms for JSON

EncodeTableKey does not work for JSON, so avoid collecting histograms
for them.

Release note: None

otan commented:
turns out we need the Key encoding because that's how ordering works. i've changed the histograms to not include the json column instead /shrug

otan commented:
(feel free to merge if it works in the morning)

mgartner commented:
Thanks @otan! I'll merge this quick-fix now and follow up with a PR that actually creates JSON histogram buckets here.

bors r+

mgartner commented:
My follow-up PR: https://github.com/cockroachdb/cockroach/pull/62173
